### PR TITLE
Use TypeScript on videopress-core

### PR DESCRIPTION
**Pull request by pgk**
Use TS for development to match existing VP package conventions.

## Proposed changes:

* Alters package devDependencies and config files.

## Jetpack product discussion

pxWta-1PG-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Running `pnpm jetpack test js-packages/videopress-core` succeeds
* Running `pnpm jetpack build js-packages/videopress-core` succeeds.

### CODE REVIEW

File: projects/js-packages/videopress-core/tests/index.test.ts [hunk 1]
```diff
@@ -6,8 +6,14 @@
...
 
+import { addition } from '../src/index';
+
 describe( 'Blank Test', () => {
 	it( 'is a noop', () => {
 		expect( true ).toBe( true );
 	} );
+
+	it( 'addition works', () => {
+		expect( addition( 1, 2 ) ).toBe( 3 );
```
Comment (roundhill) [inline projects/js-packages/videopress-core/tests/index.test.ts:17]: Why do we need this test? 🤔

**Review by roundhill — DISMISSED** [review]
All the test steps passed for me. Do we need to worry about the e2e test failing? 🤔

**Comment by jeherve** [comment]
> Do we need to worry about the e2e test failing? 🤔

No, it isn't related to this PR. That test is a bit flaky at the moment. :\

File: projects/js-packages/videopress-core/tests/index.test.ts (hunk 1 above)
Comment (pgk) [reply projects/js-packages/videopress-core/tests/index.test.ts:17]: Going to remove it in the next pr

**Review by bindlegirl — APPROVED** [review]
Tests passed

**Pull request merged**